api-schema/xdb.yaml

```diff
@@ -414,6 +427,105 @@ components:
           $ref: '#/components/schemas/ProcessIdReusePolicy'
+        primaryGlobalAttribute: 
+          $ref: '#/components/schemas/PrimaryGlobalAttribute'
+        upsertGlobalAttributes:  # initial upsert global attribute -- if the row of the primaryGlobalAttribute doesn't exist, this is to create it  
+          type: array
+          items:
+            $ref: '#/components/schemas/GlobalAttributeValue'
+    PrimaryGlobalAttribute:
+      type: object
+      description: primary global attribute is for locating the row of the database table for other global attributes, like a primary key of a table
+      required: 
+        - primaryKey
+        - primaryTable
+      properties:
+        primaryKey:
+          $ref: '#/components/schemas/GlobalAttributeValue'
+        primaryTable:
+          type: string
+          description: the primary table name of the primary global attribute
+    GlobalAttributeValue:
+      type: object
+      description: the value of a global attribute (from SDK to server or from server to SDK)
+      required: 
+        - dbColumn
+        - dbQueryValue
+      properties:
+        dbColumn:
+          type: string
+          description: the column name that can be used in the database query, see below for example
+        dbQueryValue:
+          type: string
+          description: the plain string value that can be used in the database query(e.g. for SQL SELECT ... WHERE $Column=$dbQueryValue or UPDATE/INSERT)
+        alternativeTable:
+          type: string
+          description: if not empty, the global attribute is mapped to a different table name than the primary table. Must be used with alternativeTableForeignKeyColumn
+        alternativeTableForeignKeyColumn:
+          type: string
+          description: Must present when alternativeTable is not empty. It means the foreign key column to join with the primary key of the primary table
+#        dbHint:
+#          type: string
+#          description: some additional info for how to use the dbColumn and dbQueryValue 
+    GlobalAttributeKey:
+      type: object
+      description: the definition(key) for getting value of a global attribute 
+      required:
+        - dbColumn
+      properties:
+        dbColumn:
+          type: string
+          description: the column name that can be used in the database query, see below for example
+        alternativeTable:
+          type: string
+          description: if not empty, the global attribute is mapped to a different table name than the primary table. Must be used with alternativeTableForeignKeyColumn
+        alternativeTableForeignKeyColumn:
+          type: string
+          description: Must present when alternativeTable is not empty. It means the foreign key column to join with the primary key of the primary table
+    #        dbHint:
+    #          type: string
+    #          description: some additional info for how to use the dbColumn and dbQueryValue     
+    GlobalAttributeRequest:
```

Comment:
Should we add the verb on so that making the name to `LoadGlobalAttributeRequest`?

Reply:
And maybe make it plural -> `LoadGlobalAttributesRequest `

Reply:
Yeah good idea